### PR TITLE
feat(sf_movies_5): Rename 'title' column of movies table to 'name' and perform zero downtime migration.

### PR DESCRIPTION
**Pull request by vinay-pad**
What: Rename title column of movies table to name and perform zero downtime migration of this change.

Why: These changes are fifth part of how to build your own api, involving performing ZDT migration for database schema changes.

Details:
Part 1
- Add a new migration that adds a new nullable column called 'name' to the movies and removes notNullable constraint from 'title' column.
- Change application code to still accept title but set the 'name' column of the DB to the 'title' passed in.

Part 2
- Write a migration to backfill all the null values of the newly added 'name' column.

Part 3
- Write a migration to add not-null constraint to 'name' column and drop the 'title' column.

### CODE REVIEW

**Comment by vinay-pad** [comment]
@tonywuhl Looks like this pull request is showing "files changed" as 17 since the sf_movies_4 branch hasn't yet been merged. The only files that changed for this branch(sf_movies_5) are,
        new file:   db/migrations/20170709132149_add_name_to_movies.js
        modified:   lib/plugins/features/movies/controller.js

Do you think i should wait to merge sf_movies_4 branch first before opening up this branch? If not you can just review the above 2 files for this branch. Thanks!

**Comment by thetonywu** [comment]
You can try changing the base for this PR to sf_movies_4. Once that branch is merged into master, you can switch the base back to master.

**Comment by vinay-pad** [comment]
Cool, just changed the base branch to sf_movies_4. The changes should now reflect only 2 files.

File: lib/plugins/features/movies/controller.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const Location = require('../../../models/location');
 const Movie    = require('../../../models/movie');
 
 exports.create = (payload) => {
+  payload.name = payload.title;
```
Comment (thetonywu) [inline lib/plugins/features/movies/controller.js:10]: You may also have to `delete payload.title`. Setting it to undefined might work too. It's not required at this point but if you delete it then you'll be testing that title can now be null

**Comment by thetonywu** [comment]
I believe you need to update the serialization code so that it returns only a "title" field which comes from either the name column or title column

**Review by thetonywu — APPROVED** [review]
lgtm

File: lib/models/movie.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ module.exports = Bookshelf.Model.extend({
   serialize: function () {
     return {
       id: this.get('id'),
-      title: this.get('title'),
+      title: this.get('title') | this.get('name'),
```
Comment (robinjoseph08) [inline lib/models/movie.js]: this should be `||` not just `|`

File: .node-version [hunk 3]
```diff
@@ -1 +1 @@
-5.2.0
+6.11.0
```
Comment (robinjoseph08) [inline .node-version:1]: you should also update the version of node [on travis](https://github.com/vinay-pad/sfmovies/blob/45669df73fe0c0b9dca00099089d42375ec1aaf8/.travis.yml#L4) to be the exact same as the one you use locally

**Review by robinjoseph08 — APPROVED** [review]
LGTM!

**Pull request merged**